pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2289,6 +2292,30 @@ type NetworkSpec struct {
 	MultiClusterService MultiClusterServiceSpec `json:"multiClusterService,omitempty"`
 }
 
+// +kubebuilder:validation:Enum="";host;multus
+type NetworkProviderType string
+
+const (
+	NetworkProviderDefault = ""
+
+	NetworkProviderHost = "host"
+
+	NetworkProviderMultus = "multus"
+)
+
+type NetworkRangesSpec struct {
+	Public  []CIDR `json:"public"`
+	Cluster []CIDR `json:"cluster"`
```

Comment:
Today users could be specifying the public/cluster CIDRs via ceph config, right? Looks good now to expose it in the cluster CR.

Reply:
Right. Currently, the multus feature knows how to figure out the CIDR from the `whereabouts` IPAM type and one or two others. But many IPAM specs don't have `range` listed, which means it can't be detected. There is also the issue that IPAM specs can list many ranges, which current code doesn't support. So this effectively acts as an overriding spec for Multus about what CIDR should be used.